packages/studio-base/src/hooks/usePanelHoverVisibility.tsx

```diff
@@ -0,0 +1,50 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import { MutableRefObject, useCallback, useEffect } from "react";
+
+import { PanelRoot } from "@foxglove/studio-base/components/PanelRoot";
+
+/**
+ * Attaches a listener to the parent panel mouse enter & mouse leave events.
+ *
+ * @param ref The element to hide and show on panel hover.
+ */
+export function usePanelHoverVisibility(
```

Comment:
```suggestion
export function usePanelMousePresence(
```

If you go with the approach of it returning a state vs changing the element directly

Reply:
Good idea. I'll change it.